### PR TITLE
[SPIR-V]  Implement insertion of 'Group and Subgroup Instructions' using builtin functions

### PR DESCRIPTION
**Pull request by VyacheslavLevytskyy**
This PR adds builtin functions to insert instructions from 'Group and Subgroup' section of the SPIR-V Specification. Corresponding tests are updated, `spirv-val` run is added where it was missed.

### CODE REVIEW

File: llvm/lib/Target/SPIRV/SPIRVBuiltins.cpp [hunk 1]
```diff
@@ -1015,7 +1015,37 @@ static bool generateGroupInst(const SPIRV::IncomingCall *Call,
...
+      return buildOpFromWrapper(MIRBuilder, GroupBuiltin->Opcode, Call,
+                                GR->getSPIRVTypeID(Call->ReturnType));
+
+    // Group Operation is a literal
+    Register GroupOpReg = Call->Arguments[1];
+    const MachineInstr *MI = getDefInstrMaybeConstant(GroupOpReg, MRI);
+    if (!MI || MI->getOpcode() != TargetOpcode::G_CONSTANT) {
+      std::string DiagMsg = std::string(Builtin->Name) +
+                            ": expect a constant value of Group Operation";
+      report_fatal_error(DiagMsg.c_str());
```
Comment (michalpaszkowski) [inline llvm/lib/Target/SPIRV/SPIRVBuiltins.cpp:1031]: I think we should be safe using [a Twine here to concat the strings](https://llvm.org/docs/ProgrammersManual.html#id16) and create the message or I might be missing something

Comment (VyacheslavLevytskyy) [reply llvm/lib/Target/SPIRV/SPIRVBuiltins.cpp:1031]: I think it's even better to make it simpler and remove the builtin name entirely

Comment (michalpaszkowski) [reply llvm/lib/Target/SPIRV/SPIRVBuiltins.cpp:1031]: I agree, easier to search in case of error also!

Comment (VyacheslavLevytskyy) [reply llvm/lib/Target/SPIRV/SPIRVBuiltins.cpp]: Fixed

**Review by michalpaszkowski — APPROVED** [review]

**Pull request merged**